src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java

```diff
@@ -21,19 +21,27 @@
 @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
 @Slf4j
 public class VMWorkloadInventoryCalculator extends AbstractVMWorkloadInventoryCalculator implements Calculator<Collection<VMWorkloadInventoryModel>> {
-
     @Override
     public Collection<VMWorkloadInventoryModel> calculate(String cloudFormsJson, Map<String, Object> headers) {
+
         manifestVersion = getManifestVersion(cloudFormsJson);
         jsonParsed = JsonPath.parse(cloudFormsJson);
         scanRunDate = getScanRunDate();
 
         List<Map> vmList = readListValuesFromExpandedEnvVarPath(VMPATH, null);
 
-        return vmList.stream().map(e -> {
-            e.put("_analysisId", headers.get(RouteBuilderExceptionHandler.ANALYSIS_ID).toString());
-            return e;
-        }).map(this::createVMWorkloadInventoryModel).collect(Collectors.toList());
+        List<VMWorkloadInventoryModel> vmWorkloadInventoryModels = vmList.stream()
+                .peek(e -> {
+                    e.put("_analysisId", headers.get(RouteBuilderExceptionHandler.ANALYSIS_ID).toString());
+                    e.put("vmEmsCluster", readValueFromExpandedEnvVarPath(VMEMSCLUSTERPATH, e));
+                    e.put("ems_cluster_id", readValueFromExpandedEnvVarPath(EMSCLUSTERIDPATH, e));
+                })
+                .peek(e -> log.debug("------- Treating Analysis {} VM :{} from {} : ", headers.get(RouteBuilderExceptionHandler.ANALYSIS_ID).toString(), vmList.indexOf(e), vmList.size()))
```

Comment:
could this `peek` be "merged" into the previous one?
Or could the previous `peek` become a `map`?

Reply:
which previous one ?
which is the benefit of becoming a map and which is the drawback of being a peek ?

Reply:
regarding merging it, well it's a matter of taste. 
Here there are 2 logics that I want to have : 1 is enriching the map , and the other is to log a debug info. And I have separated to show the 2 logics.

Reply:
If you think this logics division is well explained having 2 `peeks` let's keep them